### PR TITLE
Bug 1901869: Recreate master-certs if deleted

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR resolves the issue where the master-certs are not re-created when deleted by:
* Writing the secret when it changes or when it was not found initially

/cc @syedriko 
/assign @vimalk78 @igor-karpukhin 

### Links
* https://bugzilla.redhat.com/show_bug.cgi?id=1901869

### CODE REVIEW

**Review by igor-karpukhin — APPROVED** [review]
/lgtm

File: pkg/k8shandler/certificates_test.go [hunk 1]
```diff
@@ -125,6 +125,20 @@ var _ = Describe("Reconciling", func() {
...
+			Expect(client.Delete(context.TODO(), masterCASecret)).Should(Succeed())
+			secret := &corev1.Secret{}
+			key := types.NamespacedName{Name: constants.MasterCASecretName, Namespace: constants.OpenshiftNS}
+			Expect(client.Get(context.TODO(), key, secret)).ShouldNot(Succeed())
+
+			//reconcile again
+			Expect(clusterRequest.CreateOrUpdateCertificates()).Should(Succeed())
+			Expect(client.Get(context.TODO(), key, secret)).Should(Succeed())
+
+			Expect(secret).Should(ContainKeys("ca.key", "ca.crt", "ca.db", "ca.serial.txt"))
```
Comment (syedriko) [inline pkg/k8shandler/certificates_test.go:139]: Should we check here that the component certs are in the secret, not just the CA?

**Review by syedriko — APPROVED** [review]

File: pkg/k8shandler/certificates.go [hunk 2]
```diff
@@ -22,22 +22,22 @@ func Syncronize(action func() error) error {
...
 		if errors.IsNotFound(err) {
-			return nil
+			return false, nil
 		}
-		return fmt.Errorf("Unable to get secret %s: %v", constants.MasterCASecretName, err)
+		return false, fmt.Errorf("Unable to get secret %s: %v", constants.MasterCASecretName, err)
 	}
 	workDir := utils.GetWorkingDir()
 	for name, value := range secret.Data {
 		if err != utils.WriteToWorkingDirFile(path.Join(workDir, name), value) {
```
Comment (syedriko) [inline pkg/k8shandler/certificates.go:35]: Is this supposed to be
```
if err = utils.WriteToWorkingDirFile(path.Join(workDir, name), value); err != nil {
```
?

**Pull request merged**